drivers/ninaw10/machine_pin_nina.c

```diff
@@ -98,4 +110,19 @@ void machine_pin_ext_config(machine_pin_obj_t *self, int mode, int value) {
     }
 }
 
+bool machine_pin_ext_is_adc(uint32_t channel) {
```

Comment:
I think this should take `machine_pin_obj_t *self`, same as the above pin_ext functions.  Then it's clear that the argument is a pin object, and this function can extract the `->id` member.